lib/email_api/emailer.py

```diff
@@ -0,0 +1,147 @@
+import os
+from typing import Tuple, Dict, List, Optional
+from smtplib import SMTP_SSL
+from email.header import Header
+from email.mime.application import MIMEApplication
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.utils import formatdate
+
+from structs.email_params import EmailParams
+from structs.smtp_account import SMTPAccount
+
+
+class Emailer:
+    """
+    Emailer class is responsible for building and sending emails.
+    """
+
+    # Holds absolute dirpath to directory where email attachements files are stored
+    EMAIL_ATTACHMENTS_ROOT_DIR = os.path.normpath(
+        os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), "../email_attachments"
+        )
+    )
```

Comment:
Any reason we're not using [Path](https://docs.python.org/3/library/pathlib.html)?